charmcraft/commands/store/__init__.py

```diff
@@ -825,3 +825,50 @@ def run(self, parsed_args):
             logger.info(
                 "Library %s sent to the store with version %d.%d",
                 lib_data.full_name, lib_data.api, lib_data.patch)
+
+
+class ListLibCommand(BaseCommand):
+    """List all libraries belonging to a charm."""
+    name = 'list-lib'
+    help_msg = "List all libraries from a charm"
+    overview = textwrap.dedent("""
+        List all libraries from a charm.
+
+        For each library, it will show the name and the api and patch versions
+        for its tip.
+    """)
+
+    def fill_parser(self, parser):
+        """Add own parameters to the general parser."""
+        parser.add_argument(
+            'name', nargs='?', help=(
+                "The name of the charm (optional, will get the name from"
+                "metadata.yaml if not given)"))
+
+    def run(self, parsed_args):
+        """Run the command."""
+        if parsed_args.name:
+            charm_name = parsed_args.name
+        else:
+            charm_name = get_name_from_metadata()
+            if charm_name is None:
+                raise CommandError(
+                    "Can't access name in 'metadata.yaml' file. The 'list-lib' command needs to "
+                    "be executed in a valid project's directory, or indicate the charm name with "
+                    "the --charm-name option.")
```

Comment:
This error message is sounding a bit strange to me, I wonder if we could use something like "... command must either be executed from a valid project directory, or specify a charm name using the --charm-name option"? @cjp256 may have a better suggestion here.